include/sys/time_units.h

```diff
@@ -112,13 +112,13 @@ static TIME_CONSTEXPR ALWAYS_INLINE uint64_t z_tmcvt(uint64_t t, uint32_t from_h
 		if (result32 && (t < BIT64(32))) {
 			return ((uint32_t)t) / (from_hz / to_hz);
 		} else {
-			return t / (from_hz / to_hz);
+			return t / (uint64_t)(from_hz / to_hz);
```

Comment:
`return t / ((uint64_t)from_hz / to_hz);`
To be honest that is just to satfisfy the rule because uint32_t / uint32_t will never lead to a wider type.

Reply:
I changed it in 2 places of the code.

Reply:
@ceolin  Please review it again.